core/src/test/scala/doric/syntax/DateColumnsSpec.scala

```diff
@@ -1,17 +1,83 @@
 package doric
 package syntax
 
+import doric.types.{DateType, SparkType}
+
 import java.sql.{Date, Timestamp}
 import java.time.{Instant, LocalDate}
 import org.scalatest.EitherValues
 import org.scalatest.matchers.should.Matchers
+import org.apache.spark.sql.{DataFrame, functions => f}
+import org.scalatest.funspec.AnyFunSpecLike
+
+import scala.reflect.{ClassTag, classTag}
+
+trait DateOperationsSpec extends AnyFunSpecLike with TypedColumnTest {
+
+  def df: DataFrame
+
+  def test[T: DateType: SparkType: ClassTag](tagName: String): Unit = {
+
+    describe(s"Date type $tagName") {
+      it(">") {
+        test[T, T, Boolean](tagName, (a, b) => a > b)
+      }
+      it(">=") {
+        test[T, T, Boolean](tagName, (a, b) => a >= b)
+      }
+      it("<") {
+        test[T, T, Boolean](tagName, (a, b) => a < b)
+      }
+      it("<=") {
+        test[T, T, Boolean](tagName, (a, b) => a <= b)
+      }
+    }
+  }
 
-import org.apache.spark.sql.{functions => f}
+  def test[T1: SparkType: ClassTag, T2: SparkType: ClassTag, O: SparkType](
+      tagName: String,
+      f: (DoricColumn[T1], DoricColumn[T2]) => DoricColumn[O]
+  ): Unit =
+    df.validateColumnType(
+      f(
+        col[T1](getName(tagName)),
+        col[T2](getName(tagName))
+      )
+    )
+
+  def getName(tagName: String, pos: Int = 1): String =
+    s"col_${tagName}_$pos"
+}
 
 class DateColumnsSpec
     extends DoricTestElements
     with EitherValues
-    with Matchers {
+    with Matchers
+    with DateOperationsSpec {
+
+  import spark.implicits._
+
+  def df: DataFrame =
+    List(
+      (
+        Date.valueOf("2022-05-27"),
+        LocalDate.parse("2022-05-27"),
+        new Timestamp(1653669106840L),
+        Instant.ofEpochMilli(1653669106840L)
+      )
+    ).toDF(
+      getName("date"),
+      getName("localDate"),
+      getName("timestamp"),
+      getName("instant")
+    )
+
+  df.show()
+
+  test[Date]("date")
+  test[LocalDate]("localDate")
```

Comment:
I'm seeing that the test is failing for spark 2.4, split this test in two, one for `Date` and `Timestamp` as you already have, and the ones for `LocalDate` and `Instant` have to go on the `spark_3.0_3.1_3.2` folder. We have a few examples doing it, but any doubt let us know! Also, if you want to try against 2.4 on your local computer, launching `sbt -DsparkVersion=2.4` or if you use IntelliJ, in sbt settings put the spark version you want to test against:
<img width="858" alt="Captura de Pantalla 2022-06-16 a las 19 03 03" src="https://user-images.githubusercontent.com/15030451/174126495-6d1753e2-4526-4b1b-9134-7e2a8bf59b75.png">

We will have to improve our documentation for new contributors 😅

Reply:
😱 I hadn't even thought I might need to run the tests differently for the different spark versions. Thanks for the heads up!

I believe I've done what you said, I just have one question on your preferences: I implemented the trait `DateOperationsSpec` which contains the testing logic for the comparison operators. We could reimplement this in the `spark_3.0_3.1_3.2` folder, however it would look identical. This has me thinking that maybe I could extract this trait out to something along the lines of `DateOperationsFixture` and create a new folder called `fixtures`. We could then use `DateOperationsFixture` in both the 2.4.x tests and the 3.x tests. Or of course, we could just leave the trait `DateOperationsSpec` and import it in `DateColumns3xSpec`. Do you have any thoughts or preferences on this? 😄

Reply:
Maybe implement the common logic in a trait, and extend it in the base `scala` folder(it runs for all the versions) for Date and Timestamp, and for the 3.x versions extend again the trait and only call for LocalDate and instant (yeah is a mess, but is the simplest way we found).

As you can imagine, the scala version in src and these execute in all versions and it is reachable from the rest of the specialized folders. And then we have the specialized folders that contain the name with what versions are going to be executed. Sbt checks the version and gets the folders it needs to build it.

Reply:
I've hopefully addressed this in https://github.com/hablapps/doric/pull/230/commits/551fdfb8fe9698e1b1684f238b0fddcee762e27e. Please let me know if you have any further comments or feedback! 😄